pkg/process/process.go

```diff
@@ -161,22 +174,29 @@ func (p *Process) StopWithSignal(signal syscall.Signal) {
 			}
 		}()
 
-		if cmd == nil || !cmd.Started() {
-			logrus.Errorf("Process Manager: cmd of %v hasn't started, no need to stop", p.Name)
+		if cmd == nil || !cmd.IsRunning() {
+			logrus.Errorf("Process Manager: cmd of %v is not running anymore, no need to stop", p.Name)
+			if p.State != StateStopped && p.State != StateError {
+				p.State = StateStopped
+			}
 			return
 		}
 
 		// no need for lock
-		logrus.Infof("Process Manager: trying to stop process %v", p.Name)
-		cmd.StopWithSignal(signal)
-		for i := 0; i < types.WaitCount; i++ {
-			if p.IsStopped() {
-				return
+		if needStop {
+			logrus.Infof("Process Manager: trying to stop process %v", p.Name)
+			cmd.StopWithSignal(signal)
+			for i := 0; i < types.WaitCount; i++ {
+				if p.IsStopped() {
+					return
+				}
+				logrus.Infof("Wait for process %v to shutdown", p.Name)
+				time.Sleep(types.WaitInterval)
```

Comment:
If I understand correctly, we think:

- The goroutine that SHOULD have successfully stopped the process got stuck here and could never finish?
- Another goroutine like this one could never be created because `p.State` was `StateStopping`.

And the crux of this intermediate solution is to allow another goroutine to be created after `DeleteTimeout` that skips straight to the kill step?

If my understanding is correct, my only suggestion is to not hard code `DeleteTimeout` directly and instead set it to `WaitInterval * WaitCount`. This way the new behavior only takes over after it seems the old expected behavior has failed.